### PR TITLE
GUNDI-3128: Webhooks Support

### PR DESCRIPTION
**Pull request by marianobrc**
### What does this PR do?
- Adds support for webhooks
- Adds utilities for generic json-to-json transformations
- Adds utilities for parsing binary data that is encoded as hex strings
- Updates self-registration to register webhook integrations in Gundi
- Adds basic test coverage for webhooks
- Adds usage instructions in the README for webhooks

#### Other improvements unrelated to webhooks
- Upgrade python version to v3.10
- Retry Redis errors in the state manager
- Allow enabling/disabled background processing

### Relevant link(s)
[GUNDI-3128](https://allenai.atlassian.net/browse/GUNDI-3128)

[GUNDI-3128]: https://allenai.atlassian.net/browse/GUNDI-3128?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,9 @@
-FROM python:3.8-slim
+FROM python:3.10-slim
 
 WORKDIR /code
 
 COPY requirements.txt .
+RUN pip install --upgrade pip
```
Comment (daniefdz) [inline docker/Dockerfile:6]: Let's merge both RUN statements.
`RUN pip install --upgrade pip && pip install -r /code/requirements.txt`

**Review by daniefdz — CHANGES_REQUESTED** [review]

File: docker/Dockerfile (hunk 1 above)
Comment (marianobrc) [reply docker/Dockerfile:6]: Good idea, thanks

**Review by daniefdz — APPROVED** [review]
infra side LGTM

File: app/services/utils.py [hunk 2]
```diff
@@ -8,3 +14,156 @@ def find_config_for_action(configurations, action_id):
...
+            "value": self.value,
+            "hex_format": self.hex_format,
+            "unpacked_data": self.unpacked_data
+        }
+
+
+Model = typing.TypeVar('Model', bound='BaseModel')
+
+
+class DyntamicFactory:
```
Comment (chrisdoehring) [inline app/services/utils.py:87]: Great find.

**Review by chrisdoehring — APPROVED** [review]

**Pull request merged**